### PR TITLE
Add decision explanations

### PR DESCRIPTION
**Pull request by jessie-ross**
**This PR adds decision explanations as per 641-gh-Automattic/experimentation-platform.**

Screenshot in slack

Part of https://github.com/Automattic/abacus/issues/524

Todo: Add the link in.

## How has this been tested?

- Automated tests that cover added/changed functionality
- Manual testing with production data (locally)

### CODE REVIEW

File: src/lib/analyses.test.ts [hunk 1]
```diff
@@ -197,17 +627,28 @@ describe('getAggregateRecommendation', () => {
...
+              },
+            },
           }),
         ],
         AnalysisStrategy.PpNaive,
       ),
     ).toEqual({
       decision: Analyses.AggregateRecommendationDecision.MissingAnalysis,
+      practicalStatisticalDifference: Analyses.PracticalStatisticalDifferenceStatus.Uncertain,
+      statisticalDifference: false,
```
Comment (aaronyan) [inline src/lib/analyses.test.ts:644]: I think we should include this test case for complete coverage:

```
practicalStatisticalDifference: Analyses.PracticalStatisticalDifferenceStatus.Uncertain,
statisticalDifference: true,
```

**Review by aaronyan — APPROVED** [review]
Looks good to me! Left a few suggestions in-line.

The PR is a bit large, so hopefully nothing slipped through the cracks during the review. Thank you for adding so much documentation and tests!

**Comment by jessie-ross** [comment]
> The PR is a bit large, so hopefully nothing slipped through the cracks during the review. Thank you for adding so much documentation and tests!

Oh sorry! it was supposed to be stacked onto the last PR - I will address your comments and wait for Yanir's

File: src/lib/analyses.ts [hunk 2]
```diff
@@ -23,6 +31,119 @@ export const RecommendationWarningToHuman = {
...
+  /**
+   * The CI only partially overlaps with ROPE (the region of practical equivalence, set by experimenters with minDifference):
+   * If the CI partially contains ROPE then there is a chance that the real value is either within ROPE (not practical) or outside (practical).
+   */
+  Uncertain = 'Uncertain',
+}
+
+interface DiffCredibleIntervalStats {
+  /**
+   * A difference is a difference only if it makes a difference:
```
[the remark this answers is not in the document]
Comment (jessie-ross) [reply src/lib/analyses.ts:56]: This actually comes from "How to lie with statistics" :)

[the remark this answers is not in the document]
Comment (jessie-ross) [reply src/lib/analyses.ts]: Wow apparently Darrel Huff, the author, was backed by big tobacco:

> Huff was later funded by the tobacco industry to publish a follow-up to his book on statistics: How to Lie with Smoking Statistics. This led to controversy and much criticism in the late 1960s.[2][4] The book was intended to be published by Macmillan, but near the end of 1968, the plans for its publication came to an abrupt halt. It was not until the Tobacco Master Settlement Agreement in 1998 that the existence of the book, letters between Huff and tobacco industry lawyers discussing it, and the entire unpublished manuscript itself became publicly available.[2]

https://en.wikipedia.org/wiki/Darrell_Huff

[the remark this answers is not in the document]
Comment (aaronyan) [reply src/lib/analyses.ts]: Haha wow, very interesting! I haven't read it yet, but I think [this is the story](https://kilthub.cmu.edu/articles/preprint/Huff_and_puff/13120292/1) behind Darrel Huff's involvement with the Tobacco industry.

File: src/components/experiments/single-view/results/ExperimentResults.test.tsx [hunk 3]
```diff
@@ -42,15 +43,37 @@ test('renders an appropriate message for "Missing Analyses" analyses state', asy
   await expect(container.textContent).toMatch('No results are available at the moment')
 })
 
-test('renders correctly for 1 analysis datapoint', async () => {
+test('renders correctly for 1 analysis datapoint, statistically insignificant', async () => {
```
Comment (yanirs) [inline src/components/experiments/single-view/results/ExperimentResults.test.tsx:46]: Subtle difference: "Absence of evidence is not evidence of absence" :slightly_smiling_face: 

```suggestion
test('renders correctly for 1 analysis datapoint, not statistically significant', async () => {
```

File: src/components/experiments/single-view/results/MetricAssignmentResults.tsx [hunk 4]
```diff
@@ -270,6 +300,64 @@ export default function MetricAssignmentResults({
...
+                  Analyses.AggregateRecommendationDecision.ManualAnalysisRequired && (
+                  <Typography variant='body1' gutterBottom>
+                    <strong> Different strategies are recommending conflicting variations! </strong>
+                  </Typography>
+                )}
+                <Typography variant='body1'>
+                  {
+                    differenceOverviewMessages[
+                      aggregateRecommendation.practicallySignificant as Analyses.PracticalSignificanceStatus
+                    ][String(aggregateRecommendation.statisticallySignificant) as 'true' | 'false']
```
Comment (yanirs) [inline src/components/experiments/single-view/results/MetricAssignmentResults.tsx:313]: Would this work?

```suggestion
][String(aggregateRecommendation.statisticallySignificant) as StringifiedStatisticalDifference]
```

File: src/components/experiments/single-view/results/MetricAssignmentResults.tsx [hunk 5]
```diff
@@ -270,6 +300,64 @@ export default function MetricAssignmentResults({
...
+                  }{' '}
+                </Typography>
+              </TableCell>
+            </TableRow>
+            <TableRow>
+              <TableCell>
+                <Typography variant='body1' gutterBottom>
+                  The absolute change in the conversion rate of{' '}
```
Comment (yanirs) [inline src/components/experiments/single-view/results/MetricAssignmentResults.tsx:321]: :bug: 

```suggestion
                  The absolute change in the {isConversion ? 'conversion rate' : 'ARPU'} of{' '}
```

File: src/components/experiments/single-view/results/MetricAssignmentResults.tsx [hunk 6]
```diff
@@ -127,6 +127,36 @@ const useStyles = makeStyles((theme: Theme) =>
...
+type StringifiedStatisticalDifference = 'true' | 'false'
+
+// Practical Difference Status -> (string) Statistical Difference -> string
+const differenceOverviewMessages: Record<
+  Analyses.PracticalSignificanceStatus,
+  Record<StringifiedStatisticalDifference, string>
+> = {
+  [Analyses.PracticalSignificanceStatus.Yes]: {
+    true: `There is high certainty that the change is pratically significant`,
+    false: `There is high certainty that the change is pratically significant`,
```
Comment (yanirs) [inline src/components/experiments/single-view/results/MetricAssignmentResults.tsx:139]: Typo + add a period for consistency.

```suggestion
    true: `There is high certainty that the change is practically significant.`,
    false: `There is high certainty that the change is practically significant.`,
```

File: src/components/experiments/single-view/results/MetricAssignmentResults.tsx [hunk 7]
```diff
@@ -127,6 +127,36 @@ const useStyles = makeStyles((theme: Theme) =>
...
+  },
+  [Analyses.PracticalSignificanceStatus.Uncertain]: {
+    true: `There is not enough certainty to draw a conclusion at this time, but the change is statistically different from zero.`,
+    false: `There is not enough certainty to draw a conclusion at this time.`,
+  },
+  [Analyses.PracticalSignificanceStatus.No]: {
+    true: `There is high certainty that the change isn't pratically significant, but the change is statistically different from zero.`,
+    false: `There is high certainty that the change isn't pratically significant.`,
```
Comment (yanirs) [inline src/components/experiments/single-view/results/MetricAssignmentResults.tsx:147]:
```suggestion
    true: `There is high certainty that the change isn't practically significant, but the change is statistically different from zero.`,
    false: `There is high certainty that the change isn't practically significant.`,
```

**Review by yanirs — APPROVED** [review]
LGTM to me too, but please fix the typos and conversion/ARPU bug :slightly_smiling_face:

File: src/components/experiments/single-view/results/ExperimentResults.test.tsx (hunk 3 above)
Comment (jessie-ross) [reply src/components/experiments/single-view/results/ExperimentResults.test.tsx]: I love that saying

**Pull request merged**